packages/angular-cli/utilities/check-package-manager.ts

```diff
@@ -0,0 +1,60 @@
+import * as chalk from 'chalk';
+import {exec} from 'child_process';
+import {CliConfig} from '../models/config';
+
+const Promise = require('../ember-cli/lib/ext/promise');
+const config = CliConfig.fromProject();
+
+let packageManager = 'npm';
+if (config && config.get('packageManager')) {
+  packageManager = config.get('packageManager');
+}
+
+export function checkYarnOrCNPM() {
+  return new Promise((resolve, reject) => {
+    Promise
+      .all([checkYarn(), checkCNPM()])
+      .then((data) => {
+        if (packageManager === 'npm') {
+          const isYarnInstalled = data[0];
+          const isCNPMInstalled = data[1];
+          if (isYarnInstalled && isCNPMInstalled) {
+            console.log(chalk.yellow('you can `ng set --global packageManager=yarn` ' +
+            'or `ng set --global packageManager=cnpm`'));
+          } else if (isYarnInstalled) {
+            console.log(chalk.yellow('you can `ng set --global packageManager=yarn`'));
+          } else if (isCNPMInstalled) {
+            console.log(chalk.yellow('you can `ng set --global packageManager=cnpm`'));
+          }
+        }
+        resolve();
+      })
+      .catch(resolve);
```

Comment:
I don't think we should swallow errors.

In case you didn't know, using a catch on a Promise and not throwing an error will just swallow the rejection. So you can do something like:

```typescript
Promise.resolve()
  .then(() => doSomething())  // this throws
  .catch(() => {});  // this will swallow the error and the promise returned here will be resolved properly.
```


Again though, if the code above generates an error I think we should propagate it back to the user.